cmd/diff.go

```diff
@@ -32,95 +40,253 @@ var diffCmd = newDiffCmd()
-			// arg to marshal text
-			conf1, err := asConf.NewASConfigFromBytes(mgmtLibLogger, f1, fmt1)
-			if err != nil {
-				return err
-			}
-
-			conf2, err := asConf.NewASConfigFromBytes(mgmtLibLogger, f2, fmt2)
-			if err != nil {
-				return err
-			}
-
-			// get flattened config maps
-			map1 := conf1.GetFlatMap()
-			map2 := conf2.GetFlatMap()
-
-			diffs := diffFlatMaps(
-				*map1,
-				*map2,
-			)
-
-			if len(diffs) > 0 {
-				fmt.Printf("Differences shown from %s to %s, '<' are from file1, '>' are from file2.\n", path1, path2)
-				fmt.Println(strings.Join(diffs, ""))
-				return errDiffConfigsDiffer
-			}
-
-			return nil
+			logger.Warn("Using legacy 'diff' subcommand. Use 'diff files' instead.")
+			return runFileDiff(cmd, args)
 		},
 	}
 
+	res.Version = VERSION
 	res.Flags().StringP("format", "F", "conf", "The format of the source file(s). Valid options are: yaml, yml, and conf.")
 
+	// Add subcommands
+	res.AddCommand(newDiffFilesCmd())
+	res.AddCommand(newDiffServerCmd())
+
 	return res
 }
 
+// newDiffFilesCmd creates the 'diff files' subcommand (the legacy default)
+func newDiffFilesCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "files [flags] <path/to/config1> <path/to/config2>",
+		Short: "Diff yaml or conf Aerospike configuration files.",
+		Long: `Diff is used to compare differences between two Aerospike configuration files.
+			It is used on two files of the same format from any format
+			supported by the asconfig tool, e.g. yaml or Aerospike config.
+			Schema validation is not performed on either file. The file names must end with
+			extensions signifying their formats, e.g. .conf or .yaml, or --format must be used.`,
+		Example: `
+			# Compare two local configuration files
+  				asconfig diff files aerospike1.conf aerospike2.conf
+			# Compare two local yaml configuration files
+				asconfig diff files --format yaml aerospike1.yaml aerospike2.yaml`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			logger.Debug("Running diff files command")
+			return runFileDiff(cmd, args)
+		},
+	}
+	cmd.Version = VERSION
+	cmd.Flags().StringP("format", "F", "conf", "The format of the source file(s). Valid options are: yaml, yml, and conf.")
+	return cmd
+}
+
+func newDiffServerCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "server [flags] <path/to/config>",
+		Short: "BETA: Diff a local config file against a running Aerospike server's configuration.",
+		Long: `BETA: Diff is used to compare a local configuration file against the configuration of a running Aerospike server. 
+				In this mode, only one config file path is required as an argument.
+				Note: The configuration file can be in yaml or conf format.`,
+		Example: `Diff a local .conf file against a running server
+  				asconfig diff server -h 127.0.0.1:3000 aerospike.conf`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			logger.Debug("Running server diff command")
+			return runServerDiff(cmd, args)
+		},
+	}
```

Comment:
```suggestion
func newDiffServerCmd() *cobra.Command {
	cmd := &cobra.Command{
		Use:   "server [flags] <path/to/config>",
		Short: "BETA: Diff a local config file against a running Aerospike server's configuration.",
		Long: `BETA: Diff is used to compare a local configuration file against the configuration of a running Aerospike server. 
				This is useful for spotting drift between expected and actual Aerospike server configurations.
				In this mode, only one config file path is required as an argument.
				Note: The configuration file can be in yaml or conf format.`,
		Example: `Diff a local .conf file against a running server
  				asconfig diff server -h 127.0.0.1:3000 aerospike.conf`,
		RunE: func(cmd *cobra.Command, args []string) error {
			logger.Debug("Running server diff command")
			return runServerDiff(cmd, args)
		},
	}
```
A little note about how to use it might be good. Use this if you want.